src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetDeserializer.cs

```diff
@@ -76,7 +76,8 @@ public override async Task<object> ReadAsync(ODataMessageReader messageReader, T
                 throw Error.Argument("edmType", SRResources.ArgumentMustBeOfType, EdmTypeKind.Complex + " or " + EdmTypeKind.Entity);
             }
 
-            ODataReader resourceSetReader = readContext.IsChangedObjectCollection ? await messageReader.CreateODataDeltaResourceSetReaderAsync() : await messageReader.CreateODataResourceSetReaderAsync();
+            // If we want to support '#$delta' context URL, then we need to pass the EntitySet and StructuredType params to the CreateODataDeltaResourceSetReaderAsync method. 
```

Comment:
Curious, is this something that can be validated with a test?

Reply:
Sure. Will have to wait for the ODL release to update these changes.